File: data/multilang_comparison/program_c/src/func6.c
#include "program_c.h" 
array_t* func6(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array159 = func7(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array159->id);
      free(params0.data);
      array159->refC--;
      if(array159->refC == 0) {
         free(array159->data);
         free(array159);
         DEBUG_FREE(array159->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array169 = func8(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array169->id);
      free(params0.data);
      array169->refC--;
      if(array169->refC == 0) {
         free(array169->data);
         free(array169);
         DEBUG_FREE(array169->id);
      }
   }
   array_t_param params0;
   params0.size = 0;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   array_t* array174 = func9(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array174->id);
   free(params0.data);
   return array174;
}

File: data/multilang_comparison/program_nim/src/func6.nim
proc func6*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   if (PATH0 and 1'u64) != 0:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array255: ptr ArrayT = func7(addr params0, getPath(), loopsFactor)
      debugReturn(array255.id)
      dealloc(params0.data)
      dec array255.refC
      if array255.refC == 0:
         dealloc(array255.data)
         dealloc(array255)
         debugFree(array255.id)
   else:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array265: ptr ArrayT = func8(addr params0, getPath(), loopsFactor)
      debugReturn(array265.id)
      dealloc(params0.data)
      dec array265.refC
      if array265.refC == 0:
         dealloc(array265.data)
         dealloc(array265)
         debugFree(array265.id)
   var params0: ArrayTParam
   params0.size = 0
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   var array270: ptr ArrayT = func9(addr params0, getPath(), loopsFactor)
   debugReturn(array270.id)
   dealloc(params0.data)
   return array270

